### PR TITLE
Verbose mode logs expensive linalg routines

### PR DESCRIPTION
**Pull request by gpleiss**
Whenever I debug, I'm always adding print statements to log CG/Cholesky/etc. calls. Seems like a useful tool to actually add to the library :)

```python
with gpytorch.settings.verbose(True):
    # Training and eval code
```

Produces the following output:

```sh
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 1/10 - Loss: 4.051 in 0.06775569915771484
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 2/10 - Loss: 3.862 in 0.06846332550048828
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 3/10 - Loss: 3.637 in 0.063079833984375
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 4/10 - Loss: 3.421 in 0.06230592727661133
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 5/10 - Loss: 3.213 in 0.06436872482299805
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 6/10 - Loss: 3.004 in 0.0612645149230957
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 7/10 - Loss: 2.796 in 0.06605362892150879
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 8/10 - Loss: 2.587 in 0.06762933731079102
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 9/10 - Loss: 2.377 in 0.060160160064697266
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
Iter 10/10 - Loss: 2.167 in 0.05991625785827637
GPyTorch training time 0.8552227020263672
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Pivoted Cholesky on a torch.Size([5, 4979, 4979]) RHS for 10 iterations.
DEBUG:root:Running Cholesky on a matrix of size torch.Size([5, 500, 500]).
DEBUG:root:Running Lanczos on a torch.Size([4979, 4979]) matrix with a torch.Size([5, 4979, 1]) RHS for 30 iterations.
predict torch.Size([5, 11620, 18]) in 0.16702032089233398
predict 2nd time torch.Size([5, 11620, 18]) in 0.0020859241485595703
Test MAE: 0.08435824513435364
predict single point torch.Size([5, 18]) in 0.0018382072448730469
```

### CODE REVIEW

File: gpytorch/settings.py [hunk 1]
```diff
@@ -661,3 +662,16 @@ class use_toeplitz(_feature_flag):
     """
 
     _default = True
+
+
+class verbose(_feature_flag):
```
Comment (Balandat) [inline gpytorch/settings.py:667]: If we target this to linalg ops, should we be more specific about the name? Like `verbose_linalg`?

File: gpytorch/settings.py [hunk 2]
```diff
@@ -661,3 +662,16 @@ class use_toeplitz(_feature_flag):
...
+    """
+    Print out information whenever running an expesnive linear algebra routine (e.g. Cholesky, CG, Lanczos, CIQ, etc.)
+    """
+
+    _default = False
+
+    def __init__(self, state=True):
+        super().__init__(state=state)
+        if state:
+            logging.basicConfig(level=logging.DEBUG)
```
Comment (Balandat) [inline gpytorch/settings.py:677]: Hmm looks like with this logic the logging level will always be DEBUG and never reset? Should this instead overwrite the `__enter__` and `__exit__` methods?

**Review by Balandat — COMMENTED** [review]
Overall I think this useful. See my comment on the un-setting of the logging level.

There are a few other places where we will want to do this: 
https://github.com/cornellius-gp/gpytorch/blob/fa9a482ffdbc3281929c4a712ce8ab338d299854/gpytorch/lazy/lazy_tensor.py#L1882 (possibly we'll want to avoid logging in the context of Kronecker stuff)

https://github.com/cornellius-gp/gpytorch/blob/fa9a482ffdbc3281929c4a712ce8ab338d299854/gpytorch/lazy/lazy_tensor.py#L459 (again possibly we'll want to avoid logging in the context of Kronecker stuff)

File: gpytorch/settings.py (hunk 2 above)
Comment (gpleiss) [reply gpytorch/settings.py:677]: Yeah that's probably a good call.

File: gpytorch/settings.py (hunk 1 above)
Comment (gpleiss) [reply gpytorch/settings.py:667]: Will change.

**Comment by gpleiss** [comment]
Updates:
1. Renamed the setting to `gpytorch.settings.verbose_linalg`
1. Created a separate logger for `verbose_linalg` mode, so we don't mess with the global logger
1. Adding logging statements to all symeig and svd calls.

**Review by Balandat — APPROVED** [review]
lgtm

**Pull request merged**